pkg/persistence/queries.go

```diff
@@ -355,6 +355,19 @@ func checkClaimKeyBan(db queryRower, identifier string) (triesRemaining int, ban
 	return triesRemaining, banDuration, nil
 }
 
+func checkHashId(db *sql.DB, identifier string) (int64, error) {
+	var count int64
+
+	row := db.QueryRow("SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ?", identifier)
```

Comment:
If a user genuinely wants to generate another one time code and the previously generated one time code has not expired yet ( i.e the one time code is not yet wiped away from the table), he will not be able to do that with this query. He will have to wait(right now it is 10 minutes) for the one time code to wipe away from the database. There are two options to it:
A) Let him wait for 10 mins or whatever is the expiration time set
B)  
1. Check, if he(provided hashid) has generated any one time code that he has already claimed?
        SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ? and one_time_code is null;
2. If COUNT of step (1) is not ZERO,  then "unauthorized" and return.
3. If COUNT of step (1) is ZERO, check if he has generated any one time code that he has NOT yet claimed?
       SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ? and one_time_code is NOT null;
4. If COUNT of step (3) is not ZERO, delete those entries from the table and generate new one time code associated with that hashId
      DELETE FROM encryption_keys WHERE hash_id = ? and one_time_code is NOT null;
      INSERT INTO encryption_keys...............
      Return the newly generated one time code
5. If COUNT of step (3) is ZERO, generate new one time code
     INSERT INTO encryption_keys...............
     Return the newly generated one time code

Reply:
Hi @simmisaini I have incorporated your fixes - let me know if you would recommend anything else